lib/web/files.go

```diff
@@ -0,0 +1,234 @@
+/*
+Copyright 2015 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package web
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/client"
+	"github.com/gravitational/teleport/lib/reversetunnel"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/sshutils/scp"
+
+	"github.com/gravitational/trace"
+	"github.com/julienschmidt/httprouter"
+	"golang.org/x/crypto/ssh"
+)
+
+// fileTransferRequest describes HTTP file transfer request
+type fileTransferRequest struct {
+	// Server describes a server to connect to (serverId|hostname[:port]).
+	Server string
+	// Login is Linux username to connect as.
+	Login string
+	// Namespace is node namespace.
+	Namespace string
+	// Cluster is the name of the remote cluster to connect to.
+	Cluster string
+	// Cluster is the name of the remote cluster to connect to.
+	RemoteFilePath string
+}
+
+// changePassword updates users password based on the old password
+func (h *Handler) uploadFile(w http.ResponseWriter, r *http.Request, p httprouter.Params, ctx *SessionContext, site reversetunnel.RemoteSite) (interface{}, error) {
+	req := fileTransferRequest{
+		Login:          p.ByName("login"),
+		Namespace:      p.ByName("namespace"),
+		Server:         p.ByName("node"),
+		RemoteFilePath: p.ByName("filepath"),
+	}
+
+	clt, err := ctx.GetUserClient(site)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	ft := fileTransfer{
+		ctx:           ctx,
+		authClient:    clt,
+		proxyHostPort: h.ProxyHostPort(),
+	}
+
+	err = ft.upload(req, r)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return ok(), nil
+}
+
+func (h *Handler) downloadFile(w http.ResponseWriter, r *http.Request, p httprouter.Params, ctx *SessionContext, site reversetunnel.RemoteSite) (interface{}, error) {
+	req := fileTransferRequest{
+		Login:          p.ByName("login"),
```

Comment:
the structure is not exported, so it does make sense to use lower case for struct member fields as well